tools/python/openlr/quality.py

```diff
@@ -143,32 +149,38 @@ def parse_segments(tree, limit):
         # within limit are considered accurate, so golden path should be equal
         # matched path.
         golden_route = parse_route(s.find('GoldenRoute')) or matched_route
-        if not matched_route and not golden_route:
-            raise ValueError('At least one of golden route or matched route should be present')
+        yield Segment(segment_id, golden_route, matched_route)
+
+def calculate(tree):
+    result = {}
+    for s in parse_segments(tree, args.limit):
         try:
-            yield Segment(segment_id, matched_route, golden_route)
+            result[s.segment_id] = common_part(s.golden_route, s.matched_route)
         except:
```

Comment:
Please describe exception and what can go wrong

Reply:
Done.